### PR TITLE
T6934: Add preshared key for zabbix-agent monitoring service

### PR DESCRIPTION
**Pull request by HollyGurza**
- Allow configure preshared key for zabbix-agent
- Added op mode command for generatre random psk secret
- Removed duplicate xml definition for psk settings

Configure authentication mode:
```
# set service monitoring zabbix-agent authentication mode
Possible completions:
   pre-shared-secret    Use a pre-shared secret key
```

Configure PSK Settings:
```
# set service monitoring zabbix-agent authentication psk
Possible completions:
   id                   ID for authentication
   secret               pre-shared secret key

```

Generate Random PSK:
```
$ generate psk random
Possible completions:
  <Enter>               Execute the current command
  size                  Key size in bytes

```

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)

* https://vyos.dev/T6934

## Smoketest result

```
vyos@vyos:~$ python3 /usr/libexec/vyos/tests/smoke/cli/test_service_monitoring_zabbix-agent.py 
test_01_zabbix_agent (__main__.TestZabbixAgent.test_01_zabbix_agent) ... ok
test_02_zabbix_agent_psk_auth (__main__.TestZabbixAgent.test_02_zabbix_agent_psk_auth) ... ok

----------------------------------------------------------------------
Ran 2 tests in 15.881s

OK
```

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [x] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [x] My commit headlines contain a valid Task id
- [x] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

**Review by sarthurdev — APPROVED** [review]

File: interface-definitions/include/auth_mode_pre_shaerd_secret.xml.i [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<!-- include start from auth_mode_pre_shaerd_secret.xml.i -->
```
Comment (sever-sever) [inline interface-definitions/include/auth_mode_pre_shaerd_secret.xml.i]: Typo `shaerd` in the xml name and include name

**Review by sever-sever — CHANGES_REQUESTED** [review]

File: interface-definitions/service_monitoring_zabbix-agent.xml.in [hunk 2]
```diff
@@ -10,6 +10,23 @@
               <priority>1280</priority>
             </properties>
             <children>
+              <node name="authentication">
+                <properties>
+                  <help>Authentication</help>
+                </properties>
+                <children>
+                  #include <include/auth_mode_pre_shaerd_secret.xml.i>
```
Comment (sever-sever) [inline interface-definitions/service_monitoring_zabbix-agent.xml.in]: `shaerd` typo

File: interface-definitions/vpn_ipsec.xml.in [hunk 3]
```diff
@@ -722,18 +722,7 @@
...
-                      <completionHelp>
-                        <list>pre-shared-secret</list>
-                      </completionHelp>
-                      <valueHelp>
-                        <format>pre-shared-secret</format>
-                        <description>Use a pre-shared secret key</description>
-                      </valueHelp>
-                    </properties>
-                  </leafNode>
+                  #include <include/auth_mode_pre_shaerd_secret.xml.i>
```
Comment (sever-sever) [inline interface-definitions/vpn_ipsec.xml.in]: `shaerd` typo

**Review by sever-sever — CHANGES_REQUESTED** [review]

File: interface-definitions/include/auth_mode_pre_shaerd_secret.xml.i (hunk 1 above)
Comment (HollyGurza) [reply interface-definitions/include/auth_mode_pre_shaerd_secret.xml.i]: fixed

File: interface-definitions/vpn_ipsec.xml.in (hunk 3 above)
Comment (HollyGurza) [reply interface-definitions/vpn_ipsec.xml.in]: fixed

File: interface-definitions/service_monitoring_zabbix-agent.xml.in (hunk 2 above)
Comment (HollyGurza) [reply interface-definitions/service_monitoring_zabbix-agent.xml.in]: fixed

File: interface-definitions/include/auth_mode_pre_shared_secret.xml.i [hunk 4]
```diff
@@ -0,0 +1,14 @@
+<!-- include start from auth_mode_pre_shared_secret.xml.i -->
```
Comment (c-po) [inline interface-definitions/include/auth_mode_pre_shared_secret.xml.i]: We mostly (99%) use hypens in include files. Can you please stick to this unwritten rule?

File: interface-definitions/include/auth_psk_id.xml.i [hunk 5]
```diff
@@ -0,0 +1,11 @@
+<!-- include start from auth_psk_id.xml.i -->
```
Comment (c-po) [inline interface-definitions/include/auth_psk_id.xml.i]: We mostly (99%) use hypens in include files. Can you please stick to this unwritten rule?

File: interface-definitions/include/auth_psk_secret.xml.i [hunk 6]
```diff
@@ -0,0 +1,14 @@
+<!-- include start from auth_psk_secret.xml.i -->
```
Comment (c-po) [inline interface-definitions/include/auth_psk_secret.xml.i]: We mostly (99%) use hypens in include files. Can you please stick to this unwritten rule?

File: interface-definitions/include/auth_psk_secret.xml.i [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+  <properties>
+    <help>pre-shared secret key</help>
+    <valueHelp>
+      <format>txt</format>
+      <description>pre-shared secret key are required to be at least 16 bytes long, which implies at least 32 characters for hexadecimal key</description>
+    </valueHelp>
+    <constraint>
+      <validator name="psk-secret"/>
+    </constraint>
+  </properties>
```
Comment (c-po) [inline interface-definitions/include/auth_psk_secret.xml.i]:
```suggestion
  <constraintErrorMessage>Pre-Shared-Keys must be at leas 16 bytes long, which implies at least 32 characterss</constraintErrorMessage>
  </properties>
```

File: interface-definitions/include/auth_psk_secret.xml.i [hunk 8]
```diff
@@ -0,0 +1,14 @@
+<!-- include start from auth_psk_secret.xml.i -->
+<leafNode name="secret">
+  <properties>
+    <help>pre-shared secret key</help>
+    <valueHelp>
+      <format>txt</format>
+      <description>pre-shared secret key are required to be at least 16 bytes long, which implies at least 32 characters for hexadecimal key</description>
```
Comment (c-po) [inline interface-definitions/include/auth_psk_secret.xml.i]:
```suggestion
      <description>16byte pre-shared-secret key (32 character hexadecimal key)</description>
```

File: smoketest/scripts/cli/test_service_monitoring_zabbix-agent.py [hunk 9]
```diff
@@ -82,6 +83,29 @@ def test_01_zabbix_agent(self):
...
+        self.assertIn('TLSPSKIdentity=smoke_test', config)
+        self.assertIn(f'TLSPSKFile={ZABBIX_PSK_FILE}', config)
+        self.assertEqual(secret, read_file(ZABBIX_PSK_FILE))
+
+        secret = '8703ce4cb3f51279acba895e1421d69d8a7e2a18546d013d564ad87ac3957f88'
+        self.cli_set(base_path + ['authentication', 'psk', 'secret', secret])
+        self.cli_commit()
+        self.assertEqual(secret, read_file(ZABBIX_PSK_FILE))
+
+
```
Comment (c-po) [inline smoketest/scripts/cli/test_service_monitoring_zabbix-agent.py:106]: Please remove additional newlines

**Review by c-po — CHANGES_REQUESTED** [review]

File: interface-definitions/include/auth_mode_pre_shared_secret.xml.i (hunk 4 above)
Comment (HollyGurza) [reply interface-definitions/include/auth_mode_pre_shared_secret.xml.i]: fixed

File: interface-definitions/include/auth_psk_id.xml.i (hunk 5 above)
Comment (HollyGurza) [reply interface-definitions/include/auth_psk_id.xml.i]: fixed

File: interface-definitions/include/auth_psk_secret.xml.i (hunk 6 above)
Comment (HollyGurza) [reply interface-definitions/include/auth_psk_secret.xml.i]: fixed

File: interface-definitions/include/auth_psk_secret.xml.i (hunk 7 above)
Comment (HollyGurza) [reply interface-definitions/include/auth_psk_secret.xml.i]: fixed

File: interface-definitions/include/auth_psk_secret.xml.i (hunk 8 above)
Comment (HollyGurza) [reply interface-definitions/include/auth_psk_secret.xml.i]: fixed

File: smoketest/scripts/cli/test_service_monitoring_zabbix-agent.py (hunk 9 above)
Comment (HollyGurza) [reply smoketest/scripts/cli/test_service_monitoring_zabbix-agent.py:106]: fixed

**Review by sever-sever — APPROVED** [review]

**Review by c-po — APPROVED** [review]

**Pull request merged**